app/src/main/java/com/hover/stax/transactions/TransactionHistoryAdapter.kt

```diff
@@ -0,0 +1,73 @@
+package com.hover.stax.transactions
+
+import com.hover.stax.utils.DateUtils.humanFriendlyDate
+import com.hover.sdk.actions.HoverAction
+import androidx.recyclerview.widget.RecyclerView
+import com.hover.stax.transactions.TransactionHistoryAdapter.HistoryViewHolder
+import android.view.ViewGroup
+import android.view.LayoutInflater
+import android.view.View
+import androidx.core.text.HtmlCompat
+import com.hover.sdk.transactions.Transaction
+import com.hover.stax.databinding.TransactionListItemBinding
+import java.util.Locale
+
+class TransactionHistoryAdapter(private var listOfTransactionActionPair: List<Pair<StaxTransaction, HoverAction?>>,
```

Comment:
I'd suggest we use a data class containing `StaxTransaction` and `HoverAction` as properties. That way we can use a `ListAdapter` here to improve performance.